### PR TITLE
server/etcdserver: fix oss-fuzz issue

### PR DESCRIPTION
**Pull request by AdamKorcz**
Fixes https://oss-fuzz.com/testcase-detail/6750173361995776.

(Oss-fuzz issue ID has not yet been assigned)

### CODE REVIEW

File: server/etcdserver/apply_v2.go [hunk 1]
```diff
@@ -123,6 +124,9 @@ func (s *EtcdServer) applyV2Request(r *RequestV2, shouldApplyV3 membership.Shoul
 		alternative: func() string { return fmt.Sprintf("id:%d,method:%s,path:%s", r.ID, r.Method, r.Path) },
 	}
 	defer func(start time.Time) {
+		if !utf8.ValidString(r.Method) {
```
Comment (serathius) [inline server/etcdserver/apply_v2.go:127]: Not sure about panic here.

Valid utf8 is just required to report metric. In normal code flow `r.Method` value that is not standard HTTP method (also not valid utf8) will result in `ErrUnknownMethod` being returned. 

Problem happens when we want to report metrics about it. As it is already invalid request, maybe it's reasonable to just skip reporting metric?

Other approach would be to check if this whole code can be deleted. For the next release we are removing all V2 API. As this file is called apply_v2.go we should check if it's still called.

**Review by serathius — APPROVED** [review]
cc @ptabor @ahrtr @spzala

**Review by ahrtr — APPROVED** [review]
LGTM

File: server/etcdserver/apply_v2.go (hunk 1 above)
Comment (spzala) [reply server/etcdserver/apply_v2.go:127]: @AdamKorcz also, isn't such a check impacts other places in the code? e.g. references to r.Method in different files, and in general.

Comment (spzala) [reply server/etcdserver/apply_v2.go:127]: @AdamKorcz ^^ :)

Comment (AdamKorcz) [reply server/etcdserver/apply_v2.go:127]: Thanks for the heads up @spzala, I will look into that this week.

**Pull request merged**